File: backend/src/main/java/com/nusmotion/backend/dto/ServiceDescription.java
package com.nusmotion.backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Route description metadata from /ServiceDescription.
 */
public record ServiceDescription(
        @JsonProperty("Route") String route,
        @JsonProperty("RouteDescription") String routeDescription,
        @JsonProperty("RouteLongName") String routeLongName
) {}

File: frontend/lib/data/models/service_description.dart
class ServiceDescription {
  final String route;
  final String routeDescription;
  final String routeLongName;

  const ServiceDescription({
    required this.route,
    required this.routeDescription,
    required this.routeLongName,
  });

  factory ServiceDescription.fromJson(Map<String, dynamic> json) =>
      ServiceDescription(
        route: json['Route'] as String? ?? '',
        routeDescription: json['RouteDescription'] as String? ?? '',
        routeLongName: json['RouteLongName'] as String? ?? '',
      );
}
